src/System.Text.JsonLab/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -383,16 +377,20 @@ private bool ReadSingleSegment()
             }
             else if (TokenType == JsonTokenType.PropertyName)
             {
-                _searchedNextLast = false;
                 return ConsumeValue(first);
             }
             else
             {
+                int prevCurrentIndex = CurrentIndex;
+                int prevPosition = _position;
+                JsonTokenType prevTokenType = TokenType;
                 if (ConsumeNextToken(first))
                 {
-                    _searchedNextLast = true;
                     return true;
                 }
+                CurrentIndex = prevCurrentIndex;
```

Comment:
Would it make sense to not store/manage position, but rather store the CurrentIndex of the last newline and then compute position by subtracting the stored value from the new CurrentIndex. This way you only need to increment CurrentIndex

Reply:
We would still need some sort of running index to implement that correctly. CurrentIndex is reset back to 0 when a new span segment is passed in so we can't subtract last newline index from CurrentIndex to get the position.